### ISSUE TITLE
Not possible to login at all once a special charactor password is used. 

### ISSUE THREAD

**Issue by shammijayasinghe** [wso2/product-apim#8999]
### Description:
Entire pack or user management seems to be getting corrupted, If we use special character passwords. 

### Steps to reproduce:

1. Use the password as mentioned in [1].
2. Change the password as follows.
```
[super_admin]
username = "foo_admin"
password = "WSO2abc321#" 
```

3. Restart the pack and check the logs. 
4. Try to login with the above credentials. 

Please refer attached doc for the error

[1] https://github.com/wso2/product-apim/issues/8998

### Affected Product Version:
APIM 3.2.0- RC2

[ErrorsAfterChangingPassword.txt](https://github.com/wso2/product-apim/files/5048145/ErrorsAfterChangingPassword.txt)

**Comment by rmsamitha** [comment]
@shammijayasinghe Did you used the password ~^&*#`43d:3; first and started pack with that and then changed the password on the deployment.toml file and started the server? Please correct me if I am wrong.
Ideally, you cannot just change the super admin password only by the config file, if the server has been started once earlier Check the 2nd note under [1].

[1] https://apim.docs.wso2.com/en/next/install-and-setup/setup/security/logins-and-passwords/maintaining-logins-and-passwords/#change-the-super-admin-credentials

**Comment by shammijayasinghe** [comment]
@rmsamitha yes, I did what you have mentioned. It seems i have not noticed the NOTE 2. I was assuming for the previous model of Add-Admin--> true in user-mgt.xml in 2.x.x versions. 

With related to this same thing, When i restarted the server after changing the pwd to the given one in #8998 , I am getting following errors. Please check on that too.

```
[2020-08-10 15:36:12,264]  INFO - CarbonUIServiceComponent API Developer Portal Default Context : https://localhost:9443/devportal
[2020-08-10 15:36:12,265]  INFO - CarbonUIServiceComponent API Publisher Default Context : https://localhost:9443/publisher
[2020-08-10 15:36:12,389] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,389] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,390] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,389] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,390] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,389] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,390] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,392] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,526] ERROR - SubscriptionDataLoaderImpl Could not retrieve subscriptions for tenantDomain : carbon.super
[2020-08-10 15:36:12,526] ERROR - SubscriptionDataLoaderImpl Could not retrieve subscriptions for tenantDomain : carbon.super
[2020-08-10 15:36:12,526] ERROR - SubscriptionDataLoaderImpl Could not retrieve subscriptions for tenantDomain : carbon.super
[2020-08-10 15:36:12,526] ERROR - SubscriptionDataLoaderImpl Could not retrieve subscriptions for tenantDomain : carbon.super
[2020-08-10 15:36:12,526] ERROR - SubscriptionDataLoaderImpl Could not retrieve subscriptions for tenantDomain : carbon.super
[2020-08-10 15:36:12,527] ERROR - SubscriptionDataLoaderImpl Could not retrieve subscriptions for tenantDomain : carbon.super
[2020-08-10 15:36:12,528] ERROR - SubscriptionDataLoaderImpl Could not retrieve subscriptions for tenantDomain : carbon.super
[2020-08-10 15:36:12,539] ERROR - SubscriptionDataStoreImpl Exception while loading Subscriptions org.wso2.carbon.apimgt.keymgt.model.exception.DataLoadingException: Error while retrieving subscription from /subscriptions
[2020-08-10 15:36:12,539] ERROR - SubscriptionDataStoreImpl Exception while loading api Policies org.wso2.carbon.apimgt.keymgt.model.exception.DataLoadingException: Error while retrieving subscription from /api-policies
[2020-08-10 15:36:12,540] ERROR - SubscriptionDataStoreImpl Exception while loading APIs org.wso2.carbon.apimgt.keymgt.model.exception.DataLoadingException: Error while retrieving subscription from /apis
[2020-08-10 15:36:12,540] ERROR - SubscriptionDataStoreImpl Exception while loading Subscription Policies org.wso2.carbon.apimgt.keymgt.model.exception.DataLoadingException: Error while retrieving subscription from /subscription-policies
[2020-08-10 15:36:12,541] ERROR - SubscriptionDataStoreImpl Exception while loading Application Policies org.wso2.carbon.apimgt.keymgt.model.exception.DataLoadingException: Error while retrieving subscription from /application-policies
[2020-08-10 15:36:12,541] ERROR - SubscriptionDataStoreImpl Exception while loading Applications org.wso2.carbon.apimgt.keymgt.model.exception.DataLoadingException: Error while retrieving subscription from /applications
[2020-08-10 15:36:12,541] ERROR - SubscriptionDataStoreImpl Exception while loading ApplicationKeyMapping org.wso2.carbon.apimgt.keymgt.model.exception.DataLoadingException: Error while retrieving subscription from /application-key-mappings
[2020-08-10 15:36:12,556] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,576] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,630] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,648] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,665] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,682] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,702] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,720] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,735] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,749] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,763] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,788] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,802] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,817] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,832] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,852] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,865] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,881] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,897] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,911] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,924] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,938] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,951] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,964] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,978] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:12,991] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,003] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,017] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,032] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,046] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,065] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,078] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,088] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,099] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,112] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,128] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,139] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,149] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,161] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,171] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,181] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,192] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,202] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,212] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,224] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,239] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,248] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,258] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
[2020-08-10 15:36:13,267] ERROR - BasicAuthenticationInterceptor Authentication failed: Invalid credentials
```

**Comment by rmsamitha** [comment]
The above error occurs due to an issue with how the CDATA wrapping and url encoding is handled for the password in the configuration model, when there is a special character in it. This is fixed via the fix made on #8998. The docs changes regarding the fix will be merged soon.

**Comment by rmsamitha** [comment]
Closing since the originally reported scenario is not a bug and the latter mentioned issue is addressed in #8998
